best-practices/tech_lead.md

```diff
@@ -0,0 +1,318 @@
+
+Zenhub performance is related to how many repos are part of a board, so it is optimal if you can keep the number small.  We've had projects with lots of repos needed, but when certain work was finished, we could ditch some of those repos and improve the performance of the zenhub board.
+
+### Not Everyone Knows Our Toolset
+
+It may be useful to walk a PO, a UX Designer, or anyone newly interacting with the team during a work cycle, through our toolset.  It may put them at ease, and could improve efficiency as well.
+
+- How we use git, github
+  - issues
+  - pull requests
+    - what are they?
+    - how to do a review, with an approve or reject or comment ...
+  - how to review code changes in a branch
+    - We can deploy a git branch to a VM for in situ vetting
+    - It needs to be coordinated (it's not necessarily just sitting there waiting)
+- How we use zenhub
+  - what are the columns?  what do they mean?
+  - how do tickets get into each column?
+  - how do we sort them?
+- How we use GoogleDocs for running notes and action Items
+  - assigning something to a person with a comment
+  - resolving the item via the comment AND with strikethrough for text
+  - making the notes text a different color from the agenda text
+
+### WorkCycle Planning
+
+__EPICS__: It's often useful to be able to articulate the work cycle goals in a small number of achievable "epics" for the work cycle. They can be specific or a bit vague.  You're looking for a short list that can be the organizing framework for a presentation slide or an elevator speech.  You might want to be able to complete this phrase for each EPIC "... and we will know this is done when x, y, z are true."
+
+Some examples:
+
+- Decouple ETD app from direct access to DOR
+- Top PO Priorities (to Improve Service Management) <-- note how this allows for a bunch of disparate things to be in a single Epic, but it's clear they have been vetted and prioritized
+- Improve UI Accessibility <-- vague as to when it is "finished" but clear as a category of work
+- Allow Multiple Moabs For a Single Druid
+- Convert "Bulk Updates" (Sync) to "Bulk Actions" (ASync)
+- Replace Cognito Authorization
+
+Note that Epics can be too vague to be useful, or too lofty to be realistic. For example, "Replace Fedora3" might be too much for a single work cycle and it may not be clear what it entails.  If so, break it down.  Maybe "Decouple App1, App2 and App3 from direct access to DOR" would be an Epic in a work cycle, or "Explore Valkyrie as a Fedora3 Alternative", with a clear idea of "and we will know this is done when we have indicated which of the N desired functionalities Valkyrie can provide."
+
+__Have Realistic Goals__
+- 5 epics, max ... preferably 3 or 4.
```

Comment:
This may be overly prescriptive since epics can vary widely in terms of scope and scale. How about something like:
> Choose an appropriate number of epics: enough to achieve critical goals but not too many that provide false hope of how much will get done.

Reply:
i don't disagree with your point, but I also think that the notion that the epics should be "rememberable" by the motivated (e.g. TL, PO, management) implies the list should be pretty short.  IMO.

Reply:
tried for some mitigating wordsmithing.

Reply:
Thank you!